enumerable_project.rb

```diff
@@ -0,0 +1,387 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+
+# rubocop:disable Metrics/PerceivedComplexity
+
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    if self.class == Array
+
+      0.upto(self.length - 1) do |i|
+        yield(self[i])
+      end
+
+    elsif self.class == Hash
+
+      for key in self
+
+        yield(key[0].to_s, key[1])
+
+      end
+
+    end
+  end
+
+  def my_each_with_index
+    if self.class == Array
+
+      0.upto(self.length - 1) do |i|
+        yield(self[i], i)
+      end
+
+    elsif self.class == Hash
+
+      i = 0
+
+      for key in self
+
+        yield([key[0].to_s, key[1]], i)
+
+        i+=1
+
+      end
+
+    end
+  end
+
+  def my_select
+    if self.class == Array
+
+      arr = []
+
+      0.upto(self.length-1) do |i|
+        arr << self[i] if yield(self[i])
+      end
+
+      return arr
+
+    elsif self.class == Hash
+
+      hash = {}
+
+      for key in self
+
+        hash[key[0]] = key[1] if yield(key[0], key[1])
+
+      end
+
+      return hash
+
+    end
+  end
+
+  def my_all?
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        return false unless yield(self[i])
+      end
+
+      return true
+
+    elsif self.class == Hash
+
+      for key in self
+
+        return false unless yield(key[0], key[1])
+
+      end
+
+      return true
+
+    end
+  end
```

Comment:
* [ ] should have working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_all? 

- [ ] when no block or argument is given returns true only when none of the collection members are false or nil, if the mentioned condition is not met it, should return false

```ruby     
[nil, false, true, []].my_all? == [nil, false, true, []].all? => false
```
- Yours is always expecting a block
![image](https://user-images.githubusercontent.com/52209067/78952748-d32dd980-7a93-11ea-8694-ca2ace567d6a.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688266-67148700-6f65-11ea-82e5-cbef16444817.png)

- [ ] when a class is passed as an argument returns true only if all of the collection is a member of such class, otherwise, it should return false

```ruby     
[1, 2.5, 'a', 9].my_all?(Integer) == [1, 2.5, 'a', 9].all?(Integer) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78952787-f193d500-7a93-11ea-892a-a47b4271f799.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688467-be1a5c00-6f65-11ea-8970-15c9e4f4913f.png)

- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex

```ruby     
%w[dog door rod blade].my_all?(/d/) == %w[dog door rod blade].all?(/d/) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78952822-08d2c280-7a94-11ea-840d-feede00d0318.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77688741-31bc6900-6f66-11ea-90ec-b6a7cee7e6ca.png)

- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern

```ruby     
[3,4,7,11].my_all?(3) == [3,4,7,11].all?(3)  => false 
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78952865-2869eb00-7a94-11ea-8ea6-02a829a77356.png)

- Real all?
![image](https://user-images.githubusercontent.com/52209067/77689001-924ba600-6f66-11ea-9b72-210b3bb1d78b.png)